src/app/modules/submission/encrypt-submission/encrypt-submission.middleware.ts

```diff
@@ -444,8 +450,71 @@ export const validateStorageSubmission = async (
         error,
       })
       req.formsg.filteredResponses = req.body.responses
+      return error
+    })
+    .andThen(({ parsedResponses, form }) => {
+      // Validate MyInfo responses
+      const { authType } = form
+      switch (authType) {
+        case FormAuthType.SGID_MyInfo:
+        case FormAuthType.MyInfo:
+          return extractMyInfoLoginJwt(req.cookies, authType)
+            .andThen(MyInfoService.verifyLoginJwt)
+            .asyncAndThen(({ uinFin }) =>
+              MyInfoService.fetchMyInfoHashes(uinFin, form._id)
+                .andThen((hashes) =>
+                  MyInfoService.checkMyInfoHashes(
+                    parsedResponses.responses,
+                    hashes,
+                  ),
+                )
+                .map<IPopulatedEmailFormWithResponsesAndHash>(
+                  (hashedFields) => ({
+                    form,
+                    hashedFields,
+                    parsedResponses,
+                  }),
+                ),
+            )
+            .mapErr((error) => {
+              logger.error({
+                message: `Error verifying MyInfo${
+                  authType === FormAuthType.SGID_MyInfo ? '(over SGID)' : ''
+                } hashes`,
+                meta: logMeta,
+                error,
+              })
+              const { statusCode, errorMessage } = mapRouteError(error)
+              return res.status(statusCode).json({
```

Comment:
We might be calling this twice, need to check this again, either with tests or manual verification.

Reply:
See L514

Reply:
you're right, modified it here [376808d](https://github.com/opengovsg/FormSG/pull/6870/commits/376808d4cd1841c81bee8be5c7462b9ae008dee4) and tested by throwing an error in `checkMyInfoHashes`